tooling/metamodel-generator/src/main/java/org/hibernate/jpamodelgen/annotation/AnnotationMetaEntity.java

```diff
@@ -283,7 +293,18 @@ else if ( containsAnnotation( method, Constants.HQL, Constants.SQL, Constants.FI
 			}
 		}
 
-		findSessionGetter( element );
+		ExecutableElement getter = findSessionGetter( element );
+		if(getter != null) {
+			if(!isPanacheType(element)) {
+				dao = true;
+				sessionType = addDaoConstructor( getter );
+			} else {
+				sessionType = getter.getReturnType().toString();
+			}
+		} else if(context.usesQuarkusOrm() || context.usesQuarkusReactive()) {
+			dao = true;
+			sessionType = setupQuarkusDaoConstructor();
+		}
```

Comment:
This logic is getting a bit messy, so could we extract a method and add some comments explaining the cases we have to handle here?